crimsobot/utils/fact_leaderboard.py

```diff
@@ -57,15 +57,34 @@ async def get_embed(self, ctx: commands.Context) -> Embed:
         else:
             for place, leader in enumerate(leaders):
                 place = self._offset + place + 1
+
+                # placeholder if subject needs to be edited; cannot be replaced in tuple
+                new_subject = None
+
+                # turn '<@[userid]>' strings into usernames e.g. @username#1234
```

Comment:
This feels like a lot of nested fors, ifs, and trys. It's not immediately clear to me how to make this simpler, or it if can be simpler*.

* - That is, without excessive Python "one-lining"

Reply:
how about... regex!

```
subject = leader.subject

def _sub_discord_username(match):
    try:
        user_id = match.group(1)
        discord_user = await ctx.bot.fetch_user(user_id)
        return f'@{str(discord_user)}'
    except Exception:
        return match.group(0)  # something happened, couldn't get the user :(

subject = re.sub(r'<@(\d+)>', _sub_discord_username, subject)
```

you'll need `import re` at the top of this file but otherwise this _should_ work

Reply:
The issue here is that we cannot use an 'await' expression inside an object function. Running it as-is:
```
SyntaxError: 'await' outside async function
```
If I naively attempt to await everything as such:
```
subject = leader.subject

async def _sub_discord_username(match):
    try:
        user_id = match.group(1)
        discord_user = await ctx.bot.fetch_user(user_id)
        return f'@{str(discord_user)}'
    except Exception:
        return match.group(0)  # something happened, couldn't get the user :(

subject = re.sub(r'<@(\d+)>', await _sub_discord_username, subject)
```
leads to this:
```
File "/mnt/c/crimsoBOT_dev/crimsoBOT/crimsobot/utils/fact_leaderboard.py", line 90, in get_embed
    subject = re.sub(r'<@(\d+)>', await _sub_discord_username, subject)
TypeError: object function can't be used in 'await' expression
```